### DISCUSSION TITLE
Scan Tables to use stream data

### DISCUSSION THREAD

**Discussion by lupe120** [lf-edge/ekuiper#1927] [Q&A]
Hello,
Is it possible to send stream data to a scan table to hold data? I see in the documentation, "Users can use table to retain a batch of data for processing". If so where/how would I access this data.

My attempt has been to to create a scan table with the following:
Table name: Testing
datasource: topic1
format: json
type: mqtt
kind: scan 

Where would I access this table "Testing"?
If I publish a payload to topic1, Would this result in the table "Testing" containing this data?

**Comment by ngjaying** [comment]
Scan table is a logical concept like the stream. Unlike lookup table, define a scan table does not have a physical "store" and you cannot query it. Like stream, scan table must be used in a rule to let it run. Please check https://ekuiper.org/docs/en/latest/guide/tables/scan.html#enrich-data

**Reply by lupe120** [reply]
Reffering to https://github.com/lf-edge/ekuiper/blob/master/docs/en_US/guide/sources/builtin/memory.md#lookup-table.

I have a stream "demo" running with a rule "rule1". "rule1" upon being triggered has a sink:memory action with a topic of "devices/device_001/messages"

I also created a look up table of "table type: memory" , with "key: id" and "data source(topic): devices/device_001/messages"

After publishing mqtt payload/message to to demo which then triggers the 'rule1' to push it to memory at topic "devices/device_001/messages". 

Would my look up table contain this message/payload?

I am seeing this in the logs:
![image](https://github.com/lf-edge/ekuiper/assets/116304794/02cfcbd8-c720-4c9a-9227-102b413a5a16)

What is the explanation for  level=error msg="add to table devices/device_001/messages omitted, value not found for key id" file="store/db.go:127"

What value is this message referring to?

**Reply by ngjaying** [reply]
Your payload does not have "id" field. Should publish data like `{"id":1, "speed":70}`

**Reply by lupe120** [reply]
How can I verify that my memory lookup table has this data?

**Reply by ngjaying** [reply]
You can create a stream to join it and check the result. Example: 
```json
{
  "id": "ruleResult",
  "sql":"SELECT * FROM demoStream CROSS JOIN yourTable",
  "actions":[
    {
      "mqtt": {
        "server": "tcp://{{yourhost}}:1883",
        "topic": "rule/result"
      }
    }
  ]
}
```
Send an event to your stream, you'll get the joined data.

There is no direct way to query now. Or you can use SQL or redis as the table source which you can query in the physical table directly.
